### PR TITLE
fix a bug in deter_sfc_gmi (#778)

### LINKED ISSUE #778: Debug gsi.x aborts in deter_sfc_gmi with invalid array index
Encounter an unexpected error while working on issue #777 .

Build debug `gsi.x` on Hera from `develop` at  [`e82365d`](https://github.com/NOAA-EMC/GSI/commit/e82365d94cac057327de439f01b8aa7793c93100).  Run 2023060712 case using files from `/scratch2/NCEPDEV/stmp3/Emily.Liu/ROTDIRS/v17allskyens`

The debug `gsi.x aborts` in `deter_sfc_mod.f90` with the message
```
132: forrtl: severe (408): fort: (3): Subscript #2 of the array ISLI_FULL has value 0 which is less than the lower bound of 1
```
The code in question is the [`isli_full(i,j)` line](https://github.com/NOAA-EMC/GSI/blob/e82365d94cac057327de439f01b8aa7793c93100/src/gsi/deter_sfc_mod.f90#L1423) below
```
     grid_dist=rearth * (rlats_sfc(klatp1) - rlats_sfc(klat1))
     n_grid=int(40000 / grid_dist) + 1
     klatn = max(klat1 - n_grid, 1)
     klonn = klon1 - n_grid
     if (klonn < 0)  klonn = nlon_sfc - klonn
     klatpn = min((klat1 + n_grid), nlat_sfc)
     klonpn = klon1 + n_grid
     if (klonpn > nlon_sfc)  klonpn = klonpn - nlon_sfc

     isflg=0
     outer: do i = klatn, klatpn
       ! assume n_grid > 2
       if (0 < klonpn - klonn .and. klonpn - klonn < nlon_sfc / 2) then
         do j = klonn, klonpn
           if (isli_full(i, j) /= 0) then
             isflg = 1
             exit outer
           end if
         end do
```
This code is in subroutine [`deter_sfc_gmi`](https://github.com/NOAA-EMC/GSI/blob/e82365d94cac057327de439f01b8aa7793c93100/src/gsi/deter_sfc_mod.f90#L1334) in file `deter_sfc_mod.f90`

Prints added to the code confirm that `klonn` is 0.  
```
132: deter_sfc_mod: grid_dist=  26045.2     rlats_sfc(p1)=-0.831898     rlats_sfc(1)=-0.835986
132: deter_sfc_mod: klonn=     0 klon1=     2 n_grid=     2 nlon_sfc=  1536
```
`j=0` is not a valid index value for array `isli_full`.

### PR DESCRIPTION
**Pull request by xincjin-NOAA**
**Description**

To fix a bug  in deter_sfc_gmi which is dissed in detail in the issue #778 Debug gsi.x aborts in deter_sfc_gmi with invalid array index

changed a check for the subscript of  an array. 

Resolves #778 

**Type of change**

Please delete options that are not relevant.

- [x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by xincjin-NOAA** [comment]
Passed ctest on Hera:

```

Test project /scratch1/NCEPDEV/da/Xin.C.Jin/GSI/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  1341.10 sec
2/6 Test #6: global_enkf ......................   Passed  2560.05 sec
3/6 Test #1: global_4denvar ...................   Passed  3304.88 sec
4/6 Test #5: hafs_3denvar_hybens ..............   Passed  3970.54 sec
5/6 Test #2: rtma .............................   Passed  4149.22 sec
6/6 Test #4: hafs_4denvar_glbens ..............   Passed  4374.31 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 4374.37 sec

```

**Comment by RussTreadon-NOAA** [comment]
The wrong fix is associated with the PR.  We want [fix @ 2a86d5b](https://github.com/NOAA-EMC/GSI-fix/tree/2a86d5b1a09a8908618d1c8a376c68e8d9b3d02c).  I don't know how `xincjin-NOAA:feature/deter_sfc_gmi` wound up with [fix @ 0be2697](https://github.com/NOAA-EMC/GSI-fix/tree/0be26971f834fe9b1d5b118e1e0ffed53facf671).  Please update the fix hash to [fix @ 2a86d5b](https://github.com/NOAA-EMC/GSI-fix/tree/2a86d5b1a09a8908618d1c8a376c68e8d9b3d02c)

**Comment by RussTreadon-NOAA** [comment]
@xincjin-NOAA , would you please run the ctests on WCOSS2 (Cactus)?

**Comment by xincjin-NOAA** [comment]
@RussTreadon-NOAA 

Ctest passed on WCOSS2 (Cactus):

```
Test project /lfs/h2/emc/da/noscrub/xin.c.jin/GSI/build
    Start 1: global_4denvar
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 2: rtma
    Start 6: global_enkf
    Start 3: rrfs_3denvar_rdasens
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  728.66 sec
2/6 Test #6: global_enkf ......................   Passed  916.55 sec
3/6 Test #2: rtma .............................   Passed  975.87 sec
4/6 Test #5: hafs_3denvar_hybens ..............   Passed  1297.47 sec
5/6 Test #4: hafs_4denvar_glbens ..............   Passed  1299.56 sec
6/6 Test #1: global_4denvar ...................   Passed  1792.74 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 1792.86 sec

```

**Comment by RussTreadon-NOAA** [comment]
Thank you @xincjin-NOAA for running ctests on Cactus and Hera.  This is sufficient.  Once we get peer reviews and approvals we can submit this PR to the GSI Handling Review team for final approval.

**Review by RussTreadon-NOAA — APPROVED** [review]
With the change from `0` to `1` in the `klonn` check, the debug `gsi.x` runs to completion for the  2023060712 case.

Approve.

**Review by CatherineThomas-NOAA — APPROVED** [review]
Looking at the code, it makes sense that this check should be excluding zero as well.  The variable isli_full is allocated as:
`allocate(isli_full(nlat_sfc,nlon_sfc))`
So a value of zero for the second index should not occur.  

Approve.

**Review by emilyhcliu — APPROVED** [review]

**Comment by RussTreadon-NOAA** [comment]
Thank you @CatherineThomas-NOAA and @emilyhcliu for your reviews and approval.

**Pull request merged**
